### PR TITLE
Split QueryTrait::eagerLoaded() into getter/setter

### PR DESCRIPTION
**Pull request by cleptric**
Continues with #9978

### CODE REVIEW

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/Datasource/QueryTrait.php [hunk 1]
```diff
@@ -181,9 +181,33 @@ public function cache($key, $config = 'default')
...
+    public function getEagerLoaded()
+    {
+        return $this->_eagerLoaded;
+    }
+
+    /**
+     * Sets the query instance to be an eager loaded query.
+     *
+     * @param bool $value Whether or not to eager load.
+     * @return $this|\Cake\ORM\Query
```
Comment (markstory) [inline src/Datasource/QueryTrait.php]: Shouldn't this be just `$this`?

Comment (cleptric) [inline src/Datasource/QueryTrait.php]: I copied it from the old method 😇
I'll remove it.

File: src/Datasource/QueryTrait.php [hunk 2]
```diff
@@ -181,9 +181,33 @@ public function cache($key, $config = 'default')
...
+     */
+    public function setEagerLoaded($value)
```
Comment (dereuromark) [inline src/Datasource/QueryTrait.php]: Dont wenn usually use enable.. and isEnabled.. instead for bool values?

**Comment by cleptric** [comment]
I removed the `setEagerLoaded()` method as we have discussed this topic here #10457

**Pull request merged**